src/components/ha-floor-picker.ts

```diff
@@ -269,216 +279,169 @@ export class HaFloorPicker extends LitElement {
         );
       }
 
-      if (!outputFloors.length) {
-        outputFloors = [
-          {
-            floor_id: NO_FLOORS_ID,
-            name: this.hass.localize("ui.components.floor-picker.no_floors"),
-            icon: null,
-            level: null,
-            aliases: [],
-            created_at: 0,
-            modified_at: 0,
-          },
-        ];
-      }
-
-      return noAdd
-        ? outputFloors
-        : [
-            ...outputFloors,
-            {
-              floor_id: ADD_NEW_ID,
-              name: this.hass.localize("ui.components.floor-picker.add_new"),
-              icon: "mdi:plus",
-              level: null,
-              aliases: [],
-              created_at: 0,
-              modified_at: 0,
-            },
-          ];
+      const items = outputFloors.map<FloorComboBoxItem>((floor) => {
+        const floorName = computeFloorName(floor);
+        return {
+          id: floor.floor_id,
+          primary: floorName,
+          floor: floor,
+          sorting_label: floor.level?.toString() || "zzzzz",
+          search_labels: [floorName, floor.floor_id, ...floor.aliases].filter(
+            (v): v is string => Boolean(v)
+          ),
+        };
+      });
+
+      return items;
     }
   );
 
-  protected updated(changedProps: PropertyValues) {
-    if (
-      (!this._init && this.hass) ||
-      (this._init && changedProps.has("_opened") && this._opened)
-    ) {
-      this._init = true;
-      const floors = this._getFloors(
-        Object.values(this.hass.floors),
-        Object.values(this.hass.areas),
-        Object.values(this.hass.devices),
-        Object.values(this.hass.entities),
-        this.includeDomains,
-        this.excludeDomains,
-        this.includeDeviceClasses,
-        this.deviceFilter,
-        this.entityFilter,
-        this.noAdd,
-        this.excludeFloors
-      ).map((floor) => ({
-        ...floor,
-        strings: [floor.floor_id, floor.name, ...floor.aliases],
-      }));
-      this.comboBox.items = floors;
-      this.comboBox.filteredItems = floors;
-    }
-  }
+  private _rowRenderer: ComboBoxLitRenderer<FloorComboBoxItem> = (item) => html`
+    <ha-combo-box-item type="button" compact>
+      ${item.icon_path
+        ? html`
+            <ha-svg-icon
+              slot="start"
+              style="margin: 0 4px"
```

Comment:
Why is this style inline? We could combine it for both icons in a class

Reply:
We can not because the row is renderer outside of this component.